Makefile

```diff
@@ -295,3 +309,40 @@ deploy: deploy-rbac deploy-crds
 ## Removes temp directories
 clean:
 	$(Q)-rm -rf ${V_FLAG} ./out
+
+.PHONY: upload-codecov-report
+# Uploads the test coverage reports to codecov.io.
+# DO NOT USE LOCALLY: must only be called by OpenShift CI when processing new PR and when a PR is merged!
+upload-codecov-report:
+	# Upload coverage to codecov.io. Since we don't run on a supported CI platform (Jenkins, Travis-ci, etc.),
+	# we need to provide the PR metadata explicitely using env vars used coming from https://github.com/openshift/test-infra/blob/master/prow/jobs.md#job-environment-variables
+	#
+	# Also: not using the `-F unittests` flag for now as it's temporarily disabled in the codecov UI
+	# (see https://docs.codecov.io/docs/flags#section-flags-in-the-codecov-ui)
+	env
+ifneq ($(PR_COMMIT), null)
+	@echo "uploading test coverage report for pull-request #$(PULL_NUMBER)..."
+	bash <(curl -s https://codecov.io/bash) \
+		-t $(CODECOV_TOKEN) \
+		-f $(GOCOV_DIR)/*.txt \
+		-C $(PR_COMMIT) \
+		-r $(REPO_OWNER)/$(REPO_NAME) \
+		-P $(PULL_NUMBER) \
+		-Z
+else
+	@echo "uploading test coverage report after PR was merged..."
+	bash <(curl -s https://codecov.io/bash) \
+		-t $(CODECOV_TOKEN) \
+		-f $(GOCOV_DIR)/*.txt \
+		-C $(BASE_COMMIT) \
+		-r $(REPO_OWNER)/$(REPO_NAME) \
+		-Z
+endif
+
+CODECOV_TOKEN := "037da9d0-5a3c-49cb-9774-f2281af621fb"
```

Comment:
Hope this isn't real. How are you replacing this in our CI , @pmacik ?

Reply:
Replacing the token with the real one is still something that I'm still working. I need to discuss it with the openshift-ci/prow team.

Reply:
@sbose78  The token in CI is passed to the test container as a secret/volumemount: https://github.com/openshift/release/pull/4556